lib/web/scripts/desktop/configure-ad.ps1

```diff
@@ -0,0 +1,170 @@
+
+
+Write-Output $TELEPORT_CA_CERT_PEM | Out-File -FilePath teleport.pem
+
+certutil -dspublish -f teleport.pem RootCA
+certutil -dspublish -f teleport.pem NTAuthCA
+certutil -pulse
+
+$ACCESS_SECURITY_TEMPLATE=@'
+[Unicode]
+Unicode=yes
+[Version]
+signature="$CHICAGO$"
+[Service General Setting]
+"SCardSvr",2,""
+'@
+
+$COMMENT_XML=@'
+<?xml version='1.0' encoding='utf-8'?>
+<policyComments xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" revision="1.0" schemaVersion="1.0" xmlns="http://www.microsoft.com/GroupPolicy/CommentDefinitions">
+  <policyNamespaces>
+    <using prefix="ns0" namespace="Microsoft.Policies.TerminalServer"></using>
+  </policyNamespaces>
+  <comments>
+    <admTemplate></admTemplate>
+  </comments>
+  <resources minRequiredRevision="1.0">
+    <stringTable></stringTable>
+  </resources>
+</policyComments>
+'@
+
+
+$ACCESS_POLICY_GUID=((Get-GPO -Name $ACCESS_GPO_NAME).Id.Guid).ToUpper()
+$ACCESS_GPO_PATH="$env:SystemRoot\SYSVOL\sysvol\$DOMAIN_NAME\Policies\{$ACCESS_POLICY_GUID}\Machine\Microsoft\Windows NT\SecEdit"
+New-Item -Type Directory -Path $ACCESS_GPO_PATH
+New-Item -Path $ACCESS_GPO_PATH -Name "GptTmpl.inf" -ItemType "file" -Value $ACCESS_SECURITY_TEMPLATE
+New-Item -Path "$env:SystemRoot\SYSVOL\sysvol\$DOMAIN_NAME\Policies\{$ACCESS_POLICY_GUID}\Machine" -Name "comment.cmtx" -ItemType "file" -Value $COMMENT_XML
+
+# Firewall
+$FIREWALL_USER_MODE_IN_TCP = "v2.31|Action=Allow|Active=TRUE|Dir=In|Protocol=6|LPort=3389|App=%SystemRoot%\system32\svchost.exe|Svc=termservice|Name=@FirewallAPI.dll,-28775|Desc=@FirewallAPI.dll,-28756|EmbedCtxt=@FirewallAPI.dll,-28752|"
+Set-GPRegistryValue -Name $ACCESS_GPO_NAME -Key "HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\WindowsFirewall" -ValueName "PolicyVersion" -Type DWORD -Value 543
+Set-GPRegistryValue -Name $ACCESS_GPO_NAME -Type String -Key "HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\WindowsFirewall\FirewallRules" -ValueName "RemoteDesktop-UserMode-In-TCP" -Value $FIREWALL_USER_MODE_IN_TCP
+
+
+# Allow remote RDP connections
+Set-GPRegistryValue -Name $ACCESS_GPO_NAME -Key "HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows NT\Terminal Services" -ValueName "fDenyTSConnections" -Type DWORD -Value 0
+Set-GPRegistryValue -Name $ACCESS_GPO_NAME -Key "HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows NT\Terminal Services" -ValueName "UserAuthentication" -Type DWORD -Value 0
+
+
+# # Step 5/7. Export your LDAP CA certificate
+certutil "-ca.cert" windows.der
+certutil -encode windows.der windows.pem
+
+gpupdate.exe /force
+
+$CA_CERT_PEM = Get-Content -Path windows.pem
+$CA_CERT_PEM_ONE_LINE = $CA_CERT_PEM -join "\n"
+
+
+$NET_BIOS_DOMAIN= (Get-ADDomain).NetBIOSName
+$LDAP_USERNAME = "$NET_BIOS_NAME\$SAM_ACCOUNT_NAME"
+
+# TODO: figure out how to get LDAP address, maybe ask user?
```

Comment:
Looks like we need to figure this out prior to merging.

Maybe this?

```
nltest /dclist:(Get-ADDomain)
```

Reply:
I've found that we can use also this:  `Resolve-DnsName -Type A $Env:COMPUTERNAME` and then use either `Name` or `Address`.

```
PS C:\Users\Administrator> Resolve-DnsName -Type A $Env:COMPUTERNAME

Name                                           Type   TTL   Section    IPAddress
----                                           ----   ---   -------    ---------
EC2AMAZ-5F5IA89.example.com                    A      1200  Question   10.0.0.41
```

The issue I'm having is that if I use `Name` then this name is most likely not resolvable if you don't add it explicitly to your DNS config or hosts file. 
In normal environment it will probably work.

The other issue is if I use the `Address` then it won't work unless I use `insecure_skip_verify: true` as I gets this LDAP error:
```
2022-08-30T13:06:26Z ERRO [WINDOWS_D] initializing LDAP client, will retry error:[
ERROR REPORT:
Original Error: *ldap.Error LDAP Result Code 200 &#34;Network Error&#34;: x509: cannot validate certificate for 10.0.0.41 because it doesn&#39;t contain any IP SANs
Stack Trace:
        /home/ec2-user/teleport/lib/srv/desktop/windows_server.go:524 github.com/gravitational/teleport/lib/srv/desktop.(*WindowsService).initializeLDAP
        /home/ec2-user/teleport/lib/srv/desktop/windows_server.go:366 github.com/gravitational/teleport/lib/srv/desktop.NewWindowsService
        /home/ec2-user/teleport/lib/service/desktop.go:198 github.com/gravitational/teleport/lib/service.(*TeleportProcess).initWindowsDesktopServiceRegistered
        /home/ec2-user/teleport/lib/service/desktop.go:51 github.com/gravitational/teleport/lib/service.(*TeleportProcess).initWindowsDesktopService.func1
        /home/ec2-user/teleport/lib/service/supervisor.go:539 github.com/gravitational/teleport/lib/service.(*LocalService).Serve
        /home/ec2-user/teleport/lib/service/supervisor.go:276 github.com/gravitational/teleport/lib/service.(*LocalSupervisor).serve.func1
        /usr/lib/golang/src/runtime/asm_amd64.s:1571 runtime.goexit
User Message: dial
        LDAP Result Code 200 &#34;Network Error&#34;: x509: cannot validate certificate for 10.0.0.80 because it doesn&#39;t contain any IP SANs
```
Not sure if I miss something in my teleport config file.